matlab/test/arrow/type/tTypeDisplay.m

```diff
@@ -0,0 +1,256 @@
+            body = body + " " + [string(type.ID); string(type.DateUnit)];
+            body = "    " + body;
+            footer = string(newline);
+            expectedDisplay = char(strjoin([header body' footer], newline));
+            actualDisplay = evalc('disp(type)');
+            testCase.verifyEqual(actualDisplay, expectedDisplay);
+        end
+
+        function Display(testCase)
+            % Verify the display of TimestampType objects.
+            %
+            % Example:
+            %
+            %  TimestampType with properties:
+            %
+            %          ID: Timestamp
+            %    TimeUnit: Second
+            %    TimeZone: "America/Anchorage"
+
+            type = arrow.timestamp(TimeUnit="Second", TimeZone="America/Anchorage"); %#ok<NASGU>
+            classnameLink = makeLinkString(FullClassName="arrow.type.TimestampType", ClassName="TimestampType", BoldFont=true);
+            header = "  " + classnameLink + " with properties:" + newline;
+            body = strjust(pad(["ID:"; "TimeUnit:"; "TimeZone:"]));
+            body = body + " " + ["Timestamp"; "Second"; """America/Anchorage"""];
+            body = "    " + body;
+            footer = string(newline);
+            expectedDisplay = char(strjoin([header body' footer], newline));
+            actualDisplay = evalc('disp(type)');
+            testCase.verifyEqual(actualDisplay, expectedDisplay);
+        end
+    end
+
+    methods
+        function verifyDisplay(testCase, actualDisplay, expectedDisplay)
+            % When the MATLAB GUI is running, '×' (char(215)) is used as
+            % the delimiter between dimension values. However, when the 
+            % GUI is not running, 'x' (char(120)) is used as the delimiter.
+            % To account for this discrepancy, check if actualDisplay 
+            % contains char(215). If not, replace all instances of
+            % char(215) in expectedDisplay with char(120).
+
+            tf = contains(actualDisplay, char(215));
+            if ~tf
+                idx = strfind(expectedDisplay, char(215));
+                expectedDisplay(idx) = char(120);
+            end
+            testCase.verifyEqual(actualDisplay, expectedDisplay);
+        end
+    end
+end
+
+function link = makeLinkString(opts)
+    arguments
+        opts.FullClassName(1, 1) string
+        opts.ClassName(1, 1) string
+        % When displaying heterogeneous arrays, only the name of the 
+        % closest shared anscestor class is displayed in bold. All other
+        % class names are not bolded.
+        opts.BoldFont(1, 1) logical
+    end
+
+    if opts.BoldFont
+        link = compose("<a href=""matlab:helpPopup %s"" style=""font-weight:bold"">%s</a>", ...
+            opts.FullClassName, opts.ClassName);
+    else
+        link = compose("<a href=""matlab:helpPopup %s"">%s</a>", opts.FullClassName, opts.ClassName);
+    end
+end
+
+function sizeString = makeDimensionString(arraySize)
```

Comment:
Sorry, I should have added this comment before. Could we rename the return value to `dimensionString`?

Reply:
ah yeah. Will do.